plugins/telemetry-otel/src/internalClusterTest/java/org/opensearch/telemetry/tracing/TelemetryTracerEnabledSanityIT.java

```diff
@@ -74,22 +70,35 @@ public void testSanityChecksWhenTracingEnabled() throws Exception {
         client.prepareSearch().setQuery(queryStringQuery("fox")).get();
         client.prepareSearch().setQuery(queryStringQuery("jumps")).get();
 
+        updateTelemetrySetting(client, false);
+        ensureGreen();
+        refresh();
+
         // Sleep for about 3s to wait for traces are published, delay is (the delay is 1s).
         Thread.sleep(3000);
 
         TelemetryValidators validators = new TelemetryValidators(
             Arrays.asList(
                 new AllSpansAreEndedProperly(),
                 new AllSpansHaveUniqueId(),
-                new NumberOfTraceIDsEqualToRequests(),
+                // TODO: We will enable this once we have Sampling support for transport Action.
```

Comment:
We should have sampling working properly, why we need to comment `NumberOfTraceIDsEqualToRequests`?

Reply:
This validates that number of request ids should be equal to the unique trace ids but there are some async scheduled spans are also there and causing this to fail. Once we have the transport action level enable/disable support the we would be enabling it just for search requests and validate it.

Reply:
> Once we have the transport action level enable/disable support the we would be enabling it just for search requests and validate it.

Be we already have that, https://github.com/opensearch-project/OpenSearch/blob/main/plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/OTelResourceProvider.java, we should be able to:
 - set probability to 0 (no sample)
 - set the `trace=true` per search request

Or there is something missed?

Reply:
Yes, I tried this but the `client.prepareSearch().setQuery(queryStringQuery("jumps")).get();` directly calling the Node client and by passing the Http layer.

Reply:
@Gaganjuneja my apologies for the. This particular test case (when uncommented), revealed another problem: the spans we create do have incorrect span kind (they all are INTERNAL). We have to fix that, for transport actions, the span kind should be set as `SERVER` (later on we will be adding the `CLIENT` ones), other like periodic tasks etc could be INTERNAL. I think we could address that in follow up pull request (not urgent).

For this test though, we could do one simple thing: we could enhance the `NumberOfTraceIDsEqualToRequests` by adding the filter by the span attributes: for example we could count all search request by collecting the spans with attribute "action=indices:data/read/search", something along these lines:

```
new NumberOfTraceIDsEqualToRequests(Attributes.of('action", "indices:data/read/search"));
```